cmd/kube-scheduler/app/options/options_test.go

```diff
@@ -442,10 +482,13 @@ pluginConfig:
 
 			// handle errors
 			if err != nil {
-				if tc.expectedError == "" {
+				switch {
+				case tc.expectedError == "" && tc.checkErrFn == nil:
 					t.Error(err)
-				} else if !strings.Contains(err.Error(), tc.expectedError) {
-					t.Errorf("expected %q, got %q", tc.expectedError, err.Error())
+				case tc.expectedError != "":
+					assert.Contains(t, err.Error(), tc.expectedError, tc.name)
+				case tc.checkErrFn != nil:
+					assert.True(t, tc.checkErrFn(err), "got error: %v", err)
```

Comment:
This switch statement will short circuit if both `expectedError` and `checkErrFn` are supplied.
In that case, `checkErrFn` will never run, even if the the `Contains` assertion succeeds.

We also can't use `fallthrough` because `case 1` does not imply `case 2`.

Consider just using if's with `return`:
_________________________
```go
if (tc.expectedError != "" || tc.checkErrFn != nil) {
	if tc.expectedError != {
		assert.Contains(t, err.Error(), tc.expectedError, tc.name)
	}
	if tc.checkErrFn != nil {
		assert.True(t, tc.checkErrFn(err), "got error: %v", err)
	}
	return
}
t.Error(err)
return
```
_________________________
```go
var expected bool
if tc.expectedError != {
	assert.Contains(t, err.Error(), tc.expectedError, tc.name)
	expected = true
}
if tc.checkErrFn != nil {
	assert.True(t, tc.checkErrFn(err), "got error: %v", err)
	expected = true
}
if ! expected {
	t.Error(err)
}
return
```

Reply:
`t.Error(err)` can also be replaced for `assert.NoError(t, err)` if desired.

Reply:
Ah true! I fixed it